contracts/src/arbitration/KlerosGovernor.sol

```diff
@@ -0,0 +1,426 @@
+     *  @param _submissionID Submission's index in the array of submitted lists of the current sesssion.
+     *  @param _listHash Hash of a withdrawing list.
+     */
+    function withdrawTransactionList(uint256 _submissionID, bytes32 _listHash) external {
+        Session storage session = sessions[sessions.length - 1];
+        Submission storage submission = submissions[session.submittedLists[_submissionID]];
+        require(block.timestamp - lastApprovalTime <= submissionTimeout / 2, "Should be in first half");
+        // This require statement is an extra check to prevent _submissionID linking to the wrong list because of index swap during withdrawal.
+        require(submission.listHash == _listHash, "Wrong list hash");
+        require(submission.submitter == msg.sender, "Only submitter can withdraw");
+        require(block.timestamp - submission.submissionTime <= withdrawTimeout, "Withdrawing time has passed.");
+        session.submittedLists[_submissionID] = session.submittedLists[session.submittedLists.length - 1];
+        session.alreadySubmitted[_listHash] = false;
+        session.submittedLists.pop();
+        session.sumDeposit = session.sumDeposit.subCap(submission.deposit);
+        payable(msg.sender).transfer(submission.deposit);
+
+        reservedETH = reservedETH.subCap(submission.deposit);
+    }
+
+    /** @dev Approves a transaction list or creates a dispute if more than one list was submitted. TRUSTED.
+     *  If nothing was submitted changes session.
+     */
+    function executeSubmissions() external duringApprovalPeriod {
+        Session storage session = sessions[sessions.length - 1];
+        require(session.status == Status.NoDispute, "Already disputed");
+        if (session.submittedLists.length == 0) {
+            lastApprovalTime = block.timestamp;
+            session.status = Status.Resolved;
+            sessions.push();
+        } else if (session.submittedLists.length == 1) {
+            Submission storage submission = submissions[session.submittedLists[0]];
+            submission.approved = true;
+            submission.approvalTime = block.timestamp;
+            uint256 sumDeposit = session.sumDeposit;
+            session.sumDeposit = 0;
+            submission.submitter.send(sumDeposit);
+            lastApprovalTime = block.timestamp;
+            session.status = Status.Resolved;
+            sessions.push();
+
+            reservedETH = reservedETH.subCap(sumDeposit);
+        } else {
+            session.status = Status.DisputeCreated;
+            uint256 arbitrationCost = arbitrator.arbitrationCost(arbitratorExtraData);
+            session.disputeID = arbitrator.createDispute{value: arbitrationCost}(
+                session.submittedLists.length,
+                arbitratorExtraData
+            );
+            session.sumDeposit = session.sumDeposit.subCap(arbitrationCost);
+
```

Comment:
missing `sessions.push();` ?

Reply:
When the list is disputed we don't push the session right away. It's one in `rule()` once the dispute is resolved